numpy/core/numeric.py

```diff
@@ -1343,11 +1343,15 @@ def roll(a, shift, axis=None):
     ----------
     a : array_like
         Input array.
-    shift : int
-        The number of places by which elements are shifted.
-    axis : int, optional
-        The axis along which elements are shifted.  By default, the array
-        is flattened before shifting, after which the original
+    shift : int or tuple of ints
+        The number of places by which elements are shifted.  If a tuple,
+        then `axis` must be a tuple of the same size, and each of the
+        given axes is shifted by the corresponding number.  If an int
```

Comment:
Coming back to the tuple-vs-sequence issue, I just noticed that `sum`, for example, would error out if a non-tuple sequence is passed as the `axis` argument.  Not sure if that means `sum` should be improved or we should only accept tuples to avoid weird edge cases...